mesonbuild/interpreter/mesonmain.py

```diff
@@ -294,21 +295,52 @@ def override_find_program_method(self, args, kwargs):
             raise InterpreterException('Second argument must be an external program or executable.')
         self.interpreter.add_find_program_override(name, exe)
 
+    @typed_kwargs('meson.override_dependency', NATIVE_KW,
+                  KwargInfo('static', bool, since='0.60.0'))
+    @typed_pos_args('meson.override_dependency', str, dependencies.Dependency)
     @FeatureNew('meson.override_dependency', '0.54.0')
-    @permittedKwargs({'native'})
     def override_dependency_method(self, args, kwargs):
```

Comment:
```suggestion
    def override_dependency_method(self, args: T.Tuple[str, dependencies.Dependency], kwargs: 'FuncOverrideDependency') -> None:
```

Plus add:
```python
from typing_extensions import TypedDict

class FuncOverrideDependency(TypedDict):
    native: mesonlib.MachineChoice
    static: T.Optional[bool]
```
inside the `if T.TYPE_CHECKING:` block.

Reply:
done.